component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/utils/TokenExchangeUtils.java

```diff
@@ -362,29 +361,17 @@ public static void setAuthorizedUserForImpersonation(OAuthTokenReqMessageContext
             throws IdentityOAuth2Exception {
 
         try {
-
-            RealmService realmService = TokenExchangeComponentServiceHolder.getInstance().getRealmService();
-
-            int tenantId = realmService.getTenantManager().getTenantId(tenantDomain);
-            org.wso2.carbon.identity.application.common.model.User user
-                    = OAuthUtil.getUserFromTenant(authenticatedSubjectIdentifier, tenantId);
-            if (user == null) {
-                throw new IdentityOAuth2ClientException(OAuth2ErrorCodes.INVALID_REQUEST,
-                        "Invalid User Id provided for Impersonation request. Unable to find the user for given " +
-                                "user id : " + authenticatedSubjectIdentifier + " tenant Domain : " + tenantDomain);
-            }
-            AuthenticatedUser authenticatedUser = new AuthenticatedUser();
-            authenticatedUser.setUserId(authenticatedSubjectIdentifier);
-            authenticatedUser.setAuthenticatedSubjectIdentifier(authenticatedSubjectIdentifier);
-            authenticatedUser.setUserName(user.getUserName());
-            authenticatedUser.setUserStoreDomain(user.getUserStoreDomain());
-            authenticatedUser.setTenantDomain(tenantDomain);
+            // Get the client ID from the claims set.
+            String clientId = tokenReqMsgCtx.getOauth2AccessTokenReqDTO().getClientId();
+            // (String userId, AuthenticatedUser impersonator, String clientId)
```

Comment:
Do we need this comment?

Reply:
Removed with 751e6830ee786f192ccbe7ea7b7592359af1c6e2